From 2b6086dfad078dedfd9075a147ed83ead68b9b6a Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Thu, 16 Jul 2020 13:07:12 +0200
Subject: [PATCH] Fix an issue with argmin with newest xarray

The modification used the xarray, rather than the numpy, version
of argmin, which is better because it refers to the named dimension
rather than the axis number.
---
 mpas_analysis/ocean/remap_depth_slices_subtask.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mpas_analysis/ocean/remap_depth_slices_subtask.py b/mpas_analysis/ocean/remap_depth_slices_subtask.py
index 6c7eb13..210a3e4 100644
--- a/mpas_analysis/ocean/remap_depth_slices_subtask.py
+++ b/mpas_analysis/ocean/remap_depth_slices_subtask.py
@@ -148,7 +148,7 @@ class RemapDepthSlicesSubtask(RemapMpasClimatologySubtask):  # {{{
                 mask[depthIndex, :] = self.maxLevelCell.values >= 0
             else:
 
-                verticalIndex = np.argmin(np.abs(zMid - depth), axis=1)
+                verticalIndex = np.abs(zMid - depth).argmin(dim='nVertLevels')
 
                 verticalIndices[depthIndex, :] = verticalIndex.values
                 mask[depthIndex, :] = np.logical_and(depth <= zTop,
-- 
2.27.0

